lang/src/org/partiql/lang/ast/ExprNodeToStatement.kt

```diff
@@ -159,7 +160,10 @@ private fun GroupBy.toAstGroupSpec(): PartiqlAst.GroupBy =
     PartiqlAst.build {
         groupBy_(
             this@toAstGroupSpec.grouping.toAstGroupStrategy(),
-            groupKeyList(this@toAstGroupSpec.groupByItems.map { groupKey_(it.expr.toAstExpr(), it.asName?.name?.asPrimitive()) }),
+            groupKeyList(this@toAstGroupSpec.groupByItems.map {
+                val keyMetas = it.asName?.metas?.toIonElementMetaContainer() ?: emptyMetaContainer()
+                groupKey_(it.expr.toAstExpr(), it.asName?.name?.asPrimitive(keyMetas) )
+            }),
```

Comment:
These changes are identical to #305's fix when converting ExprNode GroupBy -> PIG GroupBy.